src/include/execution/vm/module_metadata.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <string>
+#include <unordered_map>
+#include <utility>
+
+namespace noisepage::execution::vm {
+
+/** Compile-time information about a module. */
+class CompileTimeModuleMetadata {
+ public:
+  /** Set the TPL in this module. */
+  void SetTPL(std::string &&tpl) { map_["tpl"] = std::move(tpl); }
+  /** Set the TBC in this module. */
+  void SetTBC(std::string &&tbc) { map_["tbc"] = std::move(tbc); }
+
+  /** @return The TPL in the module, if it exists. Otherwise empty string. */
+  std::string GetTPL() const { return map_.find("tpl") == map_.end() ? "" : map_.at("tpl"); }
+  /** @return The TBC in the module, if it exists. Otherwise empty string. */
+  std::string GetTBC() const { return map_.find("tbc") == map_.end() ? "" : map_.at("tbc"); }
+
+ private:
+  // TODO(WAN): The value-type may change in the future depending on what tagged dictionaries need.
```

Comment:
@turingcompl33t Any thoughts here?

Reply:
I don't know how query ids work right now.

And to check that I understand this
CompileTimeModuleMetadata: {"tags": query id}
ExecutionProfilingLayer: {query id -> tagging dict}
?